bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/ArrayValueImpl.java

```diff
@@ -1069,9 +1069,8 @@ private void addToStringArray(ArrayValue vals, int startIndex) {
 
     private void addToByteArray(ArrayValue vals, int startIndex) {
         int endIndex = startIndex + vals.size();
-        byte[] bytes = vals.getBytes();
         for (int i = startIndex, j = 0; i < endIndex; i++, j++) {
-            this.byteValues[i] = bytes[j];
+            add(i, vals.getByte(j));
```

Comment:
This change is no longer there since I've updated the `unshift()` logic and removed these methods. https://github.com/ballerina-platform/ballerina-lang/pull/22143/files#diff-9a885c0bc35f0041691897a17bf73b0dR960

Anyway, the reason why we needed this change was that the previous logic assumed `vals` was a `byte[]`.

```java
    private void addToByteArray(ArrayValue vals, int startIndex) {
        int endIndex = startIndex + vals.size();
        byte[] bytes = vals.getBytes();
        for (int i = startIndex, j = 0; i < endIndex; i++, j++) {
            this.byteValues[i] = bytes[j];
        }
    }
```

But this may not be the case, it could be a ref array depending on how it was created. 

`getByte()` handles this for us.

The other methods such as `addToIntArray`, `addToFloatArray`, etc. were already calling `getInt()`, `getFloat()`, etc. It was only for bytes that this was not handled.